es5-shim.js

```diff
@@ -1007,6 +1007,44 @@ defineProperties($Object, {
 // ====
 //
 
+var hasNegtiveMonthYearBug = new Date(-3509827334573292).getUTCMonth() === 12;
+if (hasNegtiveMonthYearBug) {
+    var originalGetFullYear = Date.prototype.getFullYear;
+    var originalGetMonth = Date.prototype.getMonth;
+    var originalGetUTCFullYear = Date.prototype.getUTCFullYear;
+    var originalGetUTCMonth = Date.prototype.getUTCMonth;
+}
+defineProperties(Date.prototype, {
```

Comment:
the `defineProperties` call must be inside the `if` if the things it depends on are defined there.